tests/test_config.yaml

```diff
@@ -0,0 +1,19 @@
+platforms:
+  inherit: true
+  build_on_all: true
```

Comment:
I guess you are misunderstanding the meaning of `default` platforms. It does not mean you only test on one single platform, it means those platforms are run with full scope, i.e. the same way we have `default: true` set in the platform yaml. If a test has filters that does not include any default platforms, it will run whatever appears in the plaform_allow. There is no change in behavior from the detauld twister configuration, here we remove default from what upstream defines and apply our own, the additional filtering is just the same as before.

If you only want to run `nrf52840dk_nrf52840`, use `-p nrf52840dk_nrf52840`

Reply:
the default behavior will be overriden using `increased_platform_scope: false`

Reply:
My bad. Indeed, It works like default platforms. I didn't look close to realize that that these were platforms from platform_allow due to no match with default.